behind-the-veil-siteroot/imports/ui/components/pages/service/ServicesPage.jsx

```diff
@@ -63,18 +66,25 @@ export const ServicesPage = () => {
     }
 
     const serviceCardList = combined.map((service) => (<ServiceCard
-            key={service._id._str}
-            serviceId={service._id._str}
-            serviceName={service.serviceName}
-            serviceDesc={service.serviceDesc}
-            servicePrice={service.servicePrice}
-            artistUsername={service.artistUsername}
-            serviceImageData={service.serviceImageData}
-            artistAlias={service.artistAlias}
-        ></ServiceCard>))
-
-    if (document.readyState === "complete") {
-        return (<WhiteBackground pageLayout={PageLayout.LARGE_CENTER}>
+        key={service._id._str}
+        serviceId={service._id._str}
+        serviceName={service.serviceName}
+        serviceDesc={service.serviceDesc}
+        servicePrice={service.servicePrice}
+        artistUsername={service.artistUsername}
+        serviceImageData={service.serviceImageData}
+        artistAlias={service.artistAlias}
+    ></ServiceCard>))
+
+    const [isLoading, setIsLoading] = useState(true);
```

Comment:
i think isLoading shouldnt be a state either. looking at the useSubscribe docs, when you call one of those isLoading functions, react will rerender whenever a loading state is updated, so you can replace the state and the useeffect with
` const isLoading = isLoadingService() || isLoadingArtists() || isLoadingImages() ` 
to get rid of unnecessary rerenders.

Reply:
changed and removed the useeffect! thank you!